src/adapter/src/coord/sequencer/inner.rs

```diff
@@ -1926,13 +1935,105 @@ impl Coordinator {
         })
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    fn peek_stage_optimize(
+        &mut self,
+        tx: ClientTransmitter<ExecuteResponse>,
+        session: Session,
+        stage: PeekStageOptimize,
+    ) {
+        // Generate data structures that can be moved to another task where we will perform possibly
+        // expensive optimizations.
+        let catalog = self.owned_catalog();
+        let compute = self
+            .controller
+            .compute
+            .instance_snapshot(stage.cluster_id)
+            .expect("compute instance does not exist");
+        let internal_cmd_tx = self.internal_cmd_tx.clone();
+        mz_ore::task::spawn(|| "optimize peek", async move {
```

Comment:
We might be better off here using `mz_ore::task::spawn_blocking`, since it doesn't seem like any of this work is async. Tokio maintains a separate thread pool that it will dynamically size up and down as demand increases and decreases, specifically for compute intensive tasks. Specifically, tasks that will block the thread for the entire duration of its runtime, which it seems like optimizing does?

Reply:
Done.